Board.java

```diff
@@ -30,33 +28,123 @@ public Board playerMakesMove(char player, int position) {
+        return getWinner() != null;
+    }
+
+
+    public boolean isGameOver() {
+        return hasWinner() || checkForDraw();
+    }
+
+    public ArrayList<String> allWinningCombinations() {
+        ArrayList<String> combinations = new ArrayList<>();
+        combinations.addAll(getAllRows());
+        combinations.addAll(getAllColumns());
+        return combinations;
+    }
+
+    public ArrayList<String> getAllRows() {
+        ArrayList<String> rows = new ArrayList<>();
+        for (int i=0; i < dimension * dimension; i+=dimension) {
+            String row = "";
+            for (int j = i; j < i + dimension; j++) {
+                row += state.charAt(j);
+            }
+            rows.add(row);
+        }
+        return rows;
+    }
+
+    public ArrayList<String> getAllDiagonals() {
+        ArrayList<String> diagonals = new ArrayList<>();
+        String diagonal = "";
+        for (int j = 0; j < dimension * dimension + 1; j+=dimension + 1) {
+            diagonal += state.charAt(j);
+        }
+        diagonals.add(diagonal);
+
+        diagonal = "";
+        for (int j = dimension - 1; j < dimension * dimension - 1; j+=dimension - 1) {
+            diagonal += state.charAt(j);
+        }
+        diagonals.add(diagonal);
+        return diagonals;
+    }
+
+    public ArrayList<String> getAllColumns() {
+        ArrayList<String> columns = new ArrayList<>();
+        for (int i=0; i < dimension; i++) {
+            String column = "";
+            for (int j=i; j < i + dimension * dimension; j+= dimension) {
+                column += state.charAt(j);
+            }
+            columns.add(column);
+        }
+        return columns;
+    }
+
+    public boolean checkRowsForWin(char player) {
+        for (String row : getAllRows()) {
+            if (containsOnlySame(row, player)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean checkDiagonalsForWin(char player) {
+        for (String row : getAllDiagonals()) {
+            if (containsOnlySame(row, player)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public boolean checkColumnsForWin(char player) {
+        for (String row : getAllColumns()) {
+            if (containsOnlySame(row, player)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
     public boolean checkForDraw() {
         return !state.contains("-");
     }
 
-    public boolean checkForWin(char player) {
-        for (int[] formation : win) {
-            int count = 0;
-            for (int i=0; i < formation.length; i++) {
-               if (state.charAt(formation[i]) == player) count += 1;
-               if (count == 3) return true;
-           }
+    public boolean checkForWin(char player, int position) {
+        return checkRowsForWin(player) || checkColumnsForWin(player);
+    }
+
+    public boolean containsOnlySame(String row, char player) {
```

Comment:
if you swapped the arguments around, this would read nicely as 'contains only the same player in row'  e.g. containsOnlySame(player, row)